src/v2/Components/Authentication/Desktop/LoginForm.tsx

```diff
@@ -17,28 +17,42 @@ import QuickInput from "v2/Components/QuickInput"
 import { Formik, FormikProps, useFormikContext } from "formik"
 import React, { Component, useEffect, useState } from "react"
 import { recaptcha } from "v2/Utils/recaptcha"
+import { isOtpError } from "v2/Components/Authentication/helpers"
 
 interface ConditionalOtpInputProps {
   error: string
 }
 
 const ConditionalOtpInput: React.FC<ConditionalOtpInputProps> = props => {
-  const [show, setShow] = useState(false)
+  const [showOtpInput, setShowOtpInput] = useState(false)
+  const [showOtpOnDemandPrompt, setShowOtpOnDemandPrompt] = useState(false)
   const {
     errors,
     values,
     handleBlur,
     handleChange,
     setTouched,
   } = useFormikContext<InputValues>()
+  const { error } = props
 
-  if (!show && props.error === "missing two-factor authentication code") {
-    setShow(true)
+  if (!showOtpInput) {
+    if (error === "missing two-factor authentication code") {
+      setShowOtpInput(true)
+    } else if (error === "missing on-demand authentication code") {
+      setShowOtpInput(true)
+      setShowOtpOnDemandPrompt(true)
+    }
```

Comment:
**question**: Could this be simplified a bit by using the new helper function defined below?

```tsx
if (!showOtpInput) {
  if (isOtpError(error)) {
    setShowOtpInput(true)
  }

  // Maybe we should add an `isMissingOnDemandOtpError` helper function to co-locate the magic strings?
  if (error === "missing on-demand authentication code") {
    setShowOtpOnDemandPrompt(true)
  }
}
```

Reply:
This makes sense, to me! Was erring on the side of being more explicit for those unfamiliar with the otp flow, but give we had already have utilized a helper, which would likely need to be reference anyways, agree it is likely a better trade off to co-locate the messages.